src/ui_basic/listselect.cc

```diff
@@ -409,6 +409,7 @@ void BaseListselect::draw(RenderTarget& dst) {
 	// draw text lines
 	const int eff_h =
 	   selection_mode_ == ListselectLayout::kDropdown ? get_inner_h() - kMargin : get_inner_h();
+	assert(eff_h < std::numeric_limits<int32_t>::max());
```

Comment:
> looks rather strange, and not at all clear what it tries to check...

I moved that on purely mechanical basis. Logically speaking it's indeed a really strange assert regardless of position, so I traced it back:

Looks like [originates](https://github.com/widelands/widelands/commit/7be3c8561b4bb0d30d9da9af7764e79904c7f333#diff-71dc5c7636714ffe452e1dca2f6397cf8e9f185b5ed319fe3ec1751bd1456b72) in 7be3c856 as a guard due to code using a mix of signed and unsigned, when `Panel::get_h()` return type [changed from int32 to uint32](https://github.com/widelands/widelands/commit/7be3c8561b4bb0d30d9da9af7764e79904c7f333#diff-8b15bc7f8399753847fe8e8ddeca1d2d12aff0a5c5de15711b5556931d1192b0):

https://github.com/widelands/widelands/blob/7be3c8561b4bb0d30d9da9af7764e79904c7f333/src/ui_basic/listselect.cc#L350

Evolved into:
https://github.com/widelands/widelands/blob/48744d64513eefc881184e137e08a0eec881d6ab/src/ui_basic/listselect.cc#L347-L350

In 18ad348e (2 Jun 2009), `Panel::get_h()` return type [was changed back to int32](https://github.com/widelands/widelands/commit/18ad348eb09399307960cd5c12cf27b8e52f928f#diff-8b15bc7f8399753847fe8e8ddeca1d2d12aff0a5c5de15711b5556931d1192b0), so the assert was modified in 39132040 (15 Jun 2009) into:
https://github.com/widelands/widelands/blob/39132040c980f7fd862c03606734272f708384b1/src/ui_basic/listselect.cc#L347-L350

Deleting, since it's clearly been rendered unnecessary since then.